goaviatrix/gateway.go

```diff
@@ -1161,3 +1173,71 @@ func (c *Client) DisableMonitorGatewaySubnets(gateway *Gateway) error {
 	}
 	return c.PostAPI(action, form, BasicCheck)
 }
+
+func (c *Client) EnableVPNConfig(gateway *Gateway, vpnConfig *VPNConfig) error {
+	action := "edit_vpn_config"
+	form := map[string]interface{}{
+		"CID":     c.CID,
+		"action":  action,
+		"command": "enable",
+		"vpc_id":  gateway.VpcID,
+		"lb_name": gateway.GwName,
+		"key":     vpnConfig.Name,
+		"value":   vpnConfig.Value,
+	}
+	return c.PostAPI(action, form, BasicCheck)
+}
+
+func (c *Client) DisableVPNConfig(gateway *Gateway, vpnConfig *VPNConfig) error {
+	action := "edit_vpn_config"
+	form := map[string]interface{}{
+		"CID":     c.CID,
+		"action":  action,
+		"command": "disable",
+		"vpc_id":  gateway.VpcID,
+		"lb_name": gateway.GwName,
+		"key":     vpnConfig.Name,
+		"value":   "600",
+	}
+	return c.PostAPI(action, form, BasicCheck)
+}
+
+func (c *Client) GetVPNConfigList(gateway *Gateway) ([]VPNConfig, error) {
+	Url, err := url.Parse(c.baseURL)
+	if err != nil {
+		return nil, errors.New(("url Parsing failed for edit_vpn_config(show)") + err.Error())
+	}
+	showVPNConfig := url.Values{}
+	showVPNConfig.Add("CID", c.CID)
+	showVPNConfig.Add("action", "edit_vpn_config")
+	showVPNConfig.Add("command", "show")
+	showVPNConfig.Add("vpc_id", gateway.VpcID)
+	showVPNConfig.Add("lb_name", gateway.GwName)
+	Url.RawQuery = showVPNConfig.Encode()
+	resp, err := c.Get(Url.String(), nil)
+	if err != nil {
+		return nil, errors.New("HTTP Get edit_vpn_config(show) failed: " + err.Error())
+	}
+	var data VPNConfigListResp
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(resp.Body)
+	bodyString := buf.String()
+	bodyIoCopy := strings.NewReader(bodyString)
+	if err = json.NewDecoder(bodyIoCopy).Decode(&data); err != nil {
+		return nil, errors.New("Json Decode edit_vpn_config(show) failed: " + err.Error() + "\n Body: " + bodyString)
+	}
+	if !data.Return {
+		return nil, errors.New("Rest API edit_vpn_config(show) Get failed: " + data.Reason)
+	}
+	return data.Results, ErrNotFound
+}
+
+func (c *Client) GetVPNConfig(vpnConfigName string, vpnConfigList []VPNConfig) *VPNConfig {
```

Comment:
This helper function should not be an exported function of the `goaviatrix` package. The `goaviatrix` package is for functions that wrap API calls. If you just want a little helper function like this then define it in the `aviatrix` package in the same file where you use it.